test/e2e/app-dir/next-after-app-deploy/index.test.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-env jest */
+import { nextTestSetup, isNextDev, isNextDeploy } from 'e2e-utils'
+import { retry } from 'next-test-utils'
+
+const runtimes = ['nodejs', 'edge']
+
+const WAIT_BEFORE_REVALIDATING = isNextDeploy ? 10_000 : 5_000
+
+// If we want to verify that `unstable_after()` ran its callback,
+// we need it to perform some kind of side effect (because it can't affect the response).
+// In other tests, we often use logs for this, but we don't have access to those in deploy tests.
+// So instead this test relies on calling `revalidatePath` inside `unstable_after`
+// to revalidate an ISR page '/timestamp/key/[key]', and then checking if the timestamp changed --
+// if it did, we successfully ran the callback (and performed a side effect).
```

Comment:
this test wasn't originally intended to test if `revalidatePath` works -- i discovered this bug _while_ writing these tests. so the explanation here may seem off topic